library/ssl_cli.c

```diff
@@ -2794,6 +2794,7 @@ static int ssl_in_server_key_exchange_parse( mbedtls_ssl_context *ssl,
      * structural change to provide default flow assumes failure
      */
     volatile int ret = 0;
+    volatile int ret_fi = MBEDTLS_ERR_PLATFORM_FAULT_DETECTED;
```

Comment:
@AndrzejKurek , why did you change  volatile `int ret = 0;` to be initialized to 0 instead of  MBEDTLS_ERR_PLATFORM_FAULT_DETECTED ? I am almost sure it was not zero in previous version

Reply:
It is and was zero - there's a comment added just above it - changing it to a different value would require significant rework of this function, and this is an even bigger risk.

Reply:
@AndrzejKurek , OK, I missed the comment. Lets commit it with 0, but i think we should investigate this function and change its' flow in the future. I don't like that function can just jump to the end, and return success :(